src/pages/workspaces/workspace/WorkspaceContainer.test.ts

```diff
@@ -0,0 +1,109 @@
+import { render, screen, within } from '@testing-library/react';
+import { h } from 'react-hyperscript-helpers';
+import { WorkspaceContainer } from 'src/pages/workspaces/workspace/WorkspaceContainer';
+import { defaultAzureWorkspace, defaultGoogleWorkspace } from 'src/testing/workspace-fixtures';
+
+// Mocking for Nav.getLink
+jest.mock('src/libs/nav', () => ({
+  ...jest.requireActual('src/libs/nav'),
+  getLink: jest.fn(() => '/'),
+}));
+// Mocking feature preview setup
+jest.mock('src/libs/feature-previews', () => ({
+  ...jest.requireActual('src/libs/feature-previews'),
+  isFeaturePreviewEnabled: jest.fn(),
+}));
+
+const mockWorkspaceMenu = jest.fn();
+jest.mock(
+  'src/pages/workspaces/workspace/WorkspaceMenu',
+  () =>
+    function (props) {
+      mockWorkspaceMenu(props);
+      return null;
+    }
+);
```

Comment:
I suspect this could be clarified like:

```suggestion
import WorkspaceMenu from 'src/pages/workspaces/workspace/WorkspaceMenu';
...
jest.mock('src/pages/workspaces/workspace/WorkspaceMenu', () => {
  return {
    default: jest.fn().mockReturnValue(null),
    __esModule: true,
  };
});
...
expect(WorkspaceMenu).toHaveBeenCalledWith(...);
```

Reply:
For some reason that doesn't work.
I suspect the original implementor of this struggled with it for a while before coming to the current solution....